### PR TITLE
@ #202 | add docs how to extend the base box with Chef on specific projects

### LINKED ISSUE #202: add docs how to extend the base box with Chef on specific projects
Each specific projects have different requirements for configuration management.

The first step is to make sure teracy-dev is minimal enough for other projects to extend the VM by their needs.

Chef is our initial attempt, others will follow.  

Need to upgrade the workflow of Chef.

### PR DESCRIPTION
**Pull request by hoatle**
connect #202

### CODE REVIEW

**Comment by hoatle** [comment]
we need:
- @hieptranquoc to review the technical content
- @hoavt to review the writing style
- @phuonglm to approve and merge when everything is ok

**Review by hoavt — COMMENTED** [review]
@hoatle please check my comment.

File: docs/extending_teracy_dev.rst [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+Installing ``ChefDK``
+---------------------
+
+To work with Chef cookbooks, we need to install ``ChefDK``. Fortunately, there is already an
+available cookbook for us to use to install ``ChefDK`` automatically on our VM:
+https://supermarket.chef.io/cookbooks/chef-dk
+
+
+Usually, we have ``dev-setup`` directory to extend ``teracy-dev``. The initial ``dev-setup`` content
```
Comment (hoatle) [inline docs/extending_teracy_dev.rst]: Usually, we have a ``dev-setup`` directory to extend ``teracy-dev``

**Comment by hoatle** [comment]
@hoavt comments applied

**Comment by hieptranquoc** [comment]
@hoatle reviewed, everything ok!

**Review by hoavt — APPROVED** [review]
It's ok to merge

**Pull request merged**